skopt/acquisition.py

```diff
@@ -264,3 +267,120 @@ def gaussian_ei(X, model, y_opt=0.0, xi=0.01, return_grad=False):
         return values, grad
 
     return values
+
+
+def gp_hedge(X, model, y_opt=0.0, acq_funcs=None, eta=1.0, gains=None, **kwargs):
+    """
+    Probabilistically choose an acquisition functions at every iteration.
+    - The gains `g_i` are initialized to zero.
+    - At every iteration,
+        - Each acquisition function is optimised independently to propose an
+          candidate point `X_i`.
+        - Out of all these candidate points, the next point `X_best` is
+          chosen by `X_i` corresponding to sampling from $softmax(\eta g_i)$
+        - After fitting the surrogate model swith `(X_best, y_best)`,
+          the gains are updated such that $g_i -= \mu(X_i)$.
+    This function returns the acquisition values of each acquisition
+    function.
+
+    Parameters
+    ----------
+    * `X` [array-like, shape=(n_samples, n_features)]:
+        Values where the acquisition function should be computed.
+
+    * `model` [sklearn estimator that implements predict with ``return_std``]:
+        The fit estimator that approximates the function through the
+        method ``predict``.
+        It should have a ``return_std`` parameter that returns the standard
+        deviation.
+
+    * `y_opt` [float, default 0]:
+        Previous minimum value which we would like to improve upon.
+
+    * `acq_funcs` [list, default ["EI", "LCB", "PI"]]:
+        List of acquisition functions to choose at every iteration. By default
+        the acquisition functions "EI", "LCB", "PI" are chosen with their default
+        values.
+
+    * `eta` [float, default 1.0]:
+        Probability of choosing the candidate point `X_i` as proposed by
+        the `i`th acquisition function is given by $softmax(eta g[i])$
+
+    * `gains` [array-like, default [0.0, 0.0, 0.0]]:
+        See above.
+    """
+    if acq_funcs is None:
+        acq_funcs = ["EI", "LCB", "PI"]
+
+    values = []
+    for acq_func in acq_funcs:
+        curr_acq_vals = _gaussian_acquisition(
+            X=X, model=model, y_opt=y_opt, acq_func=acq_func)
+        values.append(curr_acq_vals)
+    return values
+
+
+def _acq_func_minimize(X, all_values, model, prev_min, acq_optimizer, acq_func,
```

Comment:
I'm not sure how to refactor this part of the code into a separate acquisition function since the optimisation happens independently, I'll revert to this to as it was before?